### PR TITLE
Refactoring and cleanup of catalog and environment

### PR DESCRIPTION
**Pull request by comonoidial**
Gets rids of lots of convoluted intermediate data structures and classes.

### CODE REVIEW

File: Code/src/main/java/nl/utwente/group10/haskell/catalog/FunctionEntry.java [hunk 1]
```diff
@@ -1,17 +1,19 @@
 package nl.utwente.group10.haskell.catalog;
 
 import nl.utwente.group10.haskell.type.Type;
-import nl.utwente.group10.haskell.typeparser.TypeBuilder;
 
 /**
  * A function entry in the Haskell catalog.
  */
-public class FunctionEntry extends Entry {
+public class FunctionEntry  implements Comparable<FunctionEntry> {
```
Comment (wandernauta) [inline Code/src/main/java/nl/utwente/group10/haskell/catalog/FunctionEntry.java]: Superfluous space.

File: Code/src/main/java/nl/utwente/group10/haskell/catalog/HaskellCatalog.java [hunk 2]
```diff
@@ -155,35 +150,39 @@ public final Env asEnvironment() {
...
     }
 
     /**
-     * Sets the type classes for this catalog. Clears out any existing information.
-     * @param entries The entries to set.
+     * Loads the given XML catalog into a document.
+     * @param XMLPath The path to the XML file.
+     * @param XSDPath The path to the XSD file.
+     * @return The document for the XML file.
+     * @throws CatalogException
```
Comment (wandernauta) [inline Code/src/main/java/nl/utwente/group10/haskell/catalog/HaskellCatalog.java]: When does it throw a CatalogException?

**Comment by comonoidial** [comment]
the space was removed in followup PR, and I only moved that function without touching its comments.

**Comment by wandernauta** [comment]
You're right, merged, thanks!

**Pull request merged**